protocol/messenger.go

```diff
@@ -1337,6 +1339,32 @@ func (m *Messenger) watchExpiredMessages() {
 	}()
 }
 
+// watchUnmutedChats regularly checks for chats that should be unmuted
+func (m *Messenger) watchUnmutedChats() {
+	m.logger.Debug("watching unmuted chats")
+	go func() {
+		for {
+			select {
+			case <-time.After(3 * time.Second): // Poll every 3 seconds
+				chats, _ := m.persistence.Chats()
```

Comment:
this should use in memory chats `m.allChats.Range...`, there's a few examples in the codebase